### PR TITLE
Fixed remaining `clippy::type_complexity` warnings by refactoring tuples into structs

### PR DESCRIPTION
**Pull request by kkysen**
Fixed remaining `clippy::type_complexity` warnings by refactoring tuples into structs, which also more explicitly names tuple fields:

```rust
struct ConvertedVariable {
    pub ty: Box<Type>,
    pub mutbl: Mutability,
    pub init: TranslationResult<WithStmts<Box<Expr>>>,
}
```

for use in `Translation::convert_variable`, and 

```rust
pub struct NamedReference<R> {
    pub lvalue: Box<Expr>,
    pub rvalue: R,
}
```

for use in `Translation::{name_reference, name_reference_write, name_reference_write_read}`.  It's generic so it can be used in all three of those methods.

I had a couple of questions:
* I `#[derive(Debug)]`ed `ConvertedVariable` because the other `Converted*` types in that file were `#[derive(Debug)]`ed, and then I did the same for `NamedReference` for consistency.  Do these need to be `#[derive(Debug)]`ed?
* Is `lvalue` and `rvalue` the best names for those fields?  I same a bunch of uses use `write` and `read`.

### CODE REVIEW

File: c2rust-transpile/src/translator/named_references.rs [hunk 1]
```diff
@@ -34,6 +34,21 @@ fn is_simple_lvalue(e: &Box<Expr>) -> bool {
     }
 }
 
+pub struct NamedReference<R> {
+    pub lvalue: Box<Expr>,
+    pub rvalue: R,
+}
+
+impl<R> NamedReference<R> {
+    pub fn map<S, F: Fn(R) -> S>(self, f: F) -> NamedReference<S> {
```
Comment (fw-immunant) [inline c2rust-transpile/src/translator/named_references.rs:44]: Rename this to `map_rhs`.

**Review by fw-immunant — APPROVED** [review]
Changes look fine, except for the `map_rhs` rename that I would like. Feel free to merge after that.

I think the `lvalue`/`rvalue` names are fine--they aren't self-explanatory, but I don't think anything short would be, and there's some contextual variation in how `NamedReference`s are allowed to be used anyway--this is just a finicky area, and I don't think `read`/`write` would be better names.

For `#[derive(Debug)]`, I don't think we need it if the code compiles without it. Are the nearby ones used by debug logging somewhere?

File: c2rust-transpile/src/translator/mod.rs [hunk 2]
```diff
@@ -2957,13 +2965,17 @@ impl<'c> Translation<'c> {
             self.convert_type(typ.ctype)?
         };
 
-        let mutbl = if typ.qualifiers.is_const {
+        let mutability = if typ.qualifiers.is_const {
```
[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-transpile/src/translator/mod.rs:2968]: Oh whoops, I thought I had renamed that back already.

File: c2rust-transpile/src/translator/named_references.rs (hunk 1 above)
Comment (kkysen) [reply c2rust-transpile/src/translator/named_references.rs]: I renamed it to `map_rvalue` to be consistent with the field name.

File: c2rust-transpile/src/translator/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kkysen) [reply c2rust-transpile/src/translator/mod.rs]: Fixed in https://github.com/immunant/c2rust/pull/467/commits/3b820e3b87d7b55ccc7fa3c15df3f53fcb0ebcad.

**Pull request merged**